scene/main/canvas_item.h

```diff
@@ -427,4 +427,10 @@ class CanvasTexture : public Texture2D {
 	~CanvasTexture();
 };
 
+#ifdef DEBUG_ENABLED
```

Comment:
This is a change, might be desired but needs to be considered

Reply:
Have you tested what happens on a release build when violating this? If it crashes or glitches instead of handling it gracefully it shouldn't be changed

Reply:
Good question. My guess is _something_ will screw up, but actually seeing crashes or glitches will be pretty difficult with a simple example.
About unexpected behavior you could make the same argument with the THREAD_GUARD though. I copied this snippet from it.

Reply:
I'd just remove that part, and make it safe, deciding to make a check debug only needs quite a bit of testing IMO, for cleaning up it's not worth the risk

Reply:
Can do. Worth revisiting in my opinion, though. "Drawing outside of draw" is _one_ highly unlikely scenario in a final project, more unlikely than most.

Reply:
It's an extremely cheap check so it'd only be worth it to remove annoying errors that the end user can't resolve

But there's cases where code behaves differently in release, so it's still reasonable to detect I'd say during development, especially if it causes glitches